lib/sdk/writer/index.js

```diff
@@ -24,10 +29,54 @@ const BlockWriteStream = require('./block-write-stream')
 const BlockReadStream = require('./block-read-stream')
 const ChecksumStream = require('./checksum-stream')
 const ProgressStream = require('./progress-stream')
+const imageStream = require('../image-stream')
+const diskpart = require('../../cli/diskpart')
+const constraints = require('../../shared/drive-constraints')
+const errors = require('../../shared/errors')
 const debug = require('debug')('etcher:writer')
-const EventEmitter = require('events').EventEmitter
 const _ = require('lodash')
 
+/* eslint-disable prefer-reflect */
+
+/**
+ * @summary Timeout, in milliseconds, to wait before unmounting on success
+ * @constant
+ * @type {Number}
+ */
+const UNMOUNT_ON_SUCCESS_TIMEOUT_MS = 2000
+
+/**
+ * @summary Helper function to run a set of async tasks in sequence
+ * @private
+ * @param {Array<Function>} tasks - set of tasks
+ * @param {Function} callback - callback(error)
+ * @example
+ * runSeries([
+ *   (next) => first(next),
+ *   (next) => second(next),
+ * ], (error) => {
+ *   // ...
+ * })
+ */
+const runSeries = (tasks, callback) => {
+  /**
+   * @summary Task runner
+   * @param {Error} [error] - error
+   * @example
+   * run()
+   */
+  const run = (error) => {
+    const task = tasks.shift()
+    if (error || task == null) {
```

Comment:
The task is actually `null` when there's no more tasks to work since it shifts tasks off of the array, and there'll be no more items in it – it's the end condition basically. Unless I'm missing the obvious that is :)

Reply:
Ah fair enough, I thought `callback` was meant to handle errors only